students/jstonham/README.md

```diff
@@ -2873,4 +2873,149 @@ Controller - Business logic.
+Unlike other tools such as Angular and Ember, where change can flow from a component's state to the user *or* from the user to a component's state, React *only* allows change to flow from a component's state to the user.
+
+React components are made up of tags.
+There can only be one outermost tag in a React component, but other tags can be nested inside.
+In the following example, the `render()` method has an outermost `div` tag with `h1` and `ul` tags nested inside, and inside the `ul` tag are 3 `li` tags:
+
+```JSX
+render() {
+        return (
+            <div className="shopping-list">
+            <h1>ShoppingList for {this.props.name}</h1>
+            <ul>
+                <li>Instagram</li>
+                <li>WhatsApp</li>
+                <li>Oculus</li>
+            </ul>
+            </div>
+        );
+    }
+  ```
+
+## JSX (JavaScript Extended)
+
+JavaScript Extended is JavaScript with HTML.
+React can take in JSX and render as HTML.
+
+The following is a React component called `Greeting`, written in JSX, that will render the greeting message in HTML.
+
+```JSX
+class Greeting extends React.Component {
+    render() {
+        const name = this.props.name
+        return <p> Hi! My name is {name} </p>
+    }
+}
+```
+
+React components only need one method, `render()`, which will return the new component in a JSX format.
+
+Due to the fact that it's a combination of JavaScript and HTML, JSX needs to be able to differentiate between its JavaScript and HTML classes.
+The way it does so is to call its JavaScript classes `class` and its HTML classes `className`.
+
+## Data Structure
+
+Computer scientists think of data structures as trees (although it's easier to think of them as root systems).
+
+Example data structure:
+
+| | | Root (main Parent) | | | |
+| ---| --- | --- | --- | --- | --- |
+| | Parent 1 | | | Parent 2 | |
+| Child 1 | Child 2 | Child 3 | Child 1 | Child 2 | Child 3 |
+
+Parent components can have multiple child components, but child components can have only one parent component.
+
+This data structure is not only for React components, but for structures such as Lists and Maps.
+
+In React, the child components do not need to know their parent component.
+The parent component will dictate their props, but the child component does not need to know where their props come from.
+
+Browsers use a similar structure, called the Document Object Model (DOM) as a reference for placing elements on a web page.
+Elements are nested within each other in a tree-like structure and the locations of where the elements are placed are referred to as nodes.
+Medium has a good [article](https://medium.com/javascript-in-plain-english/how-to-traverse-the-dom-in-javascript-d6555c335b4e) that explains it all quite nicely.
+
+## State and Props (Properties)
+
+A component's state is defined within the component itself.
+A component's props are defined by other components.
+(A component cannot define its own properties).
```

Comment:
I would be a bit more specific here and instead of using 'define' I suggest we use 'declare' and 'assign' perhaps.

A component 'declares' which props it will use internally by attempting to retrieve it from `this.props`. There might not be a defined behaviour of how the component will work if it was not given a prop it expected. 
If a parent gives a child a prop that the child never retrieves from `this.props` then nothing bad will happen. It will simply not use it.

A parent component 'assigns' props to all its children components by passing them in as JSX "attributes".